MantidPlot/src/PythonScript.cpp

```diff
@@ -658,9 +657,9 @@ bool PythonScript::executeString() {
   } else {
     emit finished(MSG_FINISHED);
     success = true;
-  }
-  if (isInteractive()) {
-    generateAutoCompleteList();
+    if (isInteractive() && !isExecuting()) {
```

Comment:
I think the `isExecuting` check here means that the code inside the `if` here will never run. I think `isExecuting` will only return false after `executeString()` has completed.

To demonstrate this first type `ws = CreateSampleWorkspace()` in to the script window and execute the code. Now on a new line start typing `ws.` and nothing happens. In the current version of Mantid the selection box of methods pops up, which is created by a call to `generateAutoCompleteList()`